### PR TITLE
Support workload granularity metrics dynamicly turn on with kmeshctl

### PR DESCRIPTION
**Pull request by LiZhenCheng9527**
**What type of PR is this?**
/kind enhancement

**What this PR does / why we need it**:
add workload metrics trigger in kmeshctl. 
Enable workload metrics on demand.
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: ctl/monitoring/monitoring.go [hunk 1]
```diff
@@ -33,8 +33,9 @@ import (
 )
 
 const (
-	patternAccesslog  = "/accesslog"
-	patternMonitoring = "/monitoring"
+	patternAccesslog       = "/accesslog"
+	patternMonitoring      = "/monitoring"
+	patternWorkloadMetrics = "/workloadMetrics"
```
Comment (hzxuzhonghu) [inline ctl/monitoring/monitoring.go:38]:
```suggestion
	patternWorkloadMetrics = "/workload_metrics"
```

IIUC, http url is not case sensitive

File: ctl/monitoring/monitoring.go [hunk 2]
```diff
@@ -49,8 +50,12 @@ kmeshctl monitoring <kmesh-daemon-pod> --accesslog enable/disable
 # Enable/Disable Kmesh's metrics and accesslog:
 kmeshctl monitoring <kmesh-daemon-pod> --all enable/disable
 
+# Enable/Disable Kmesh's workload metrics:
```
Comment (hzxuzhonghu) [inline ctl/monitoring/monitoring.go:53]: More detailes comments

workload metrics means what

File: ctl/monitoring/monitoring.go [hunk 3]
```diff
@@ -59,6 +64,7 @@ kmeshctl monitoring --all enable/disable`,
 	}
 	cmd.Flags().String("accesslog", "", "Control accesslog enable or disable")
 	cmd.Flags().String("all", "", "Control accesslog and metrics enable or disable together")
+	cmd.Flags().String("workloadMetrics", "", "Control Metrics for workload enable or disable")
```
Comment (hzxuzhonghu) [inline ctl/monitoring/monitoring.go:67]: more description too

File: ctl/monitoring/monitoring.go [hunk 4]
```diff
@@ -70,7 +76,8 @@ func ControlMonitoring(cmd *cobra.Command, args []string) {
 	}
 	accesslogFlag, _ := cmd.Flags().GetString("accesslog")
 	allFlag, _ := cmd.Flags().GetString("all")
-	if accesslogFlag == "" && allFlag == "" {
+	workloadMetricsFlag, _ := cmd.Flags().GetString("workloadMetrics")
+	if accesslogFlag == "" && allFlag == "" && workloadMetricsFlag == "" {
 		log.Print("no parameters. Need --accesslog or --all")
```
Comment (YaoZengzeng) [inline ctl/monitoring/monitoring.go:81]:
```suggestion
		log.Print("no parameters. Need --accesslog, --workloadMetrics or --all")
```

File: docs/ctl/kmeshctl_monitoring.md [hunk 5]
```diff
@@ -12,20 +12,25 @@ kmeshctl monitoring [flags]
...
 ### Options
 
 ```
-      --accesslog string   Control accesslog enable or disable
-      --all string         Control accesslog and metrics enable or disable together
-  -h, --help               help for monitoring
+      --accesslog string         Control accesslog enable or disable
+      --all string               Control accesslog and metrics(to-service and to-pod) enable or disable together
+  -h, --help                     help for monitoring
+      --workloadMetrics string   Control metrics to connections between pods enable or disable
```
Comment (YaoZengzeng) [inline docs/ctl/kmeshctl_monitoring.md:33]:
```suggestion
      --workloadMetrics string   Control metrics of connections between pods enable or disable
```

File: ctl/monitoring/monitoring.go [hunk 6]
```diff
@@ -63,8 +63,8 @@ kmeshctl monitoring --all enable/disable`,
 		},
 	}
 	cmd.Flags().String("accesslog", "", "Control accesslog enable or disable")
-	cmd.Flags().String("all", "", "Control accesslog and metrics enable or disable together")
-	cmd.Flags().String("workloadMetrics", "", "Control Metrics for workload enable or disable")
+	cmd.Flags().String("all", "", "Control accesslog and metrics(to-service and to-pod) enable or disable together")
+	cmd.Flags().String("workloadMetrics", "", "Control metrics to connections between pods enable or disable")
```
Comment (hzxuzhonghu) [inline ctl/monitoring/monitoring.go:67]: IIUC, when disbaled, it still record metrics with service granularity

Comment (LiZhenCheng9527) [reply ctl/monitoring/monitoring.go]: If `--all` is  enable, `--workloadMetrics` is disable, it  still record metrics with service granularity.

File: docs/ctl/kmeshctl_monitoring.md (hunk 5 above)
Comment (LiZhenCheng9527) [reply docs/ctl/kmeshctl_monitoring.md]: Done

File: ctl/monitoring/monitoring.go [hunk 7]
```diff
@@ -46,19 +47,24 @@ func NewCmd() *cobra.Command {
 		Example: `# Enable/Disable Kmesh's accesslog:
 kmeshctl monitoring <kmesh-daemon-pod> --accesslog enable/disable
 
-# Enable/Disable Kmesh's metrics and accesslog:
+# Enable/Disable Kmesh's metrics(to-service and to-pod) and accesslog:
```
Comment (hzxuzhonghu) [inline ctl/monitoring/monitoring.go:50]:
```suggestion
# Enable/Disable services' metrics and accesslog generated from bpf:
```

File: ctl/monitoring/monitoring.go [hunk 8]
```diff
@@ -46,19 +47,24 @@ func NewCmd() *cobra.Command {
...
 kmeshctl monitoring <kmesh-daemon-pod> --all enable/disable
 
+# Enable/Disable Kmesh's metrics of connections between pods:
```
Comment (hzxuzhonghu) [inline ctl/monitoring/monitoring.go:53]:
```suggestion
# Enable/Disable workload granularity metrics:
```

Comment (hzxuzhonghu) [reply ctl/monitoring/monitoring.go:53]: what does it mean if <kmesh-daemon-pod> is omitted

File: ctl/monitoring/monitoring.go [hunk 9]
```diff
@@ -46,19 +47,24 @@ func NewCmd() *cobra.Command {
...
+kmeshctl monitoring <kmesh-daemon-pod> --workloadMetrics enable/disable
+
 # If you want to change the monitoring functionality of all kmesh daemons in the cluster
 kmeshctl monitoring --accesslog enable/disable
+kmeshctl monitoring --workloadMetrics enable/disable
```
Comment (hzxuzhonghu) [inline ctl/monitoring/monitoring.go:58]: we can add these documents above together with per node control

File: ctl/monitoring/monitoring.go [hunk 10]
```diff
@@ -46,19 +47,24 @@ func NewCmd() *cobra.Command {
...
 kmeshctl monitoring --all enable/disable`,
 		Args: cobra.MaximumNArgs(1),
 		Run: func(cmd *cobra.Command, args []string) {
 			ControlMonitoring(cmd, args)
 		},
 	}
 	cmd.Flags().String("accesslog", "", "Control accesslog enable or disable")
-	cmd.Flags().String("all", "", "Control accesslog and metrics enable or disable together")
+	cmd.Flags().String("all", "", "Control accesslog and metrics(to-service and to-pod) enable or disable together")
```
Comment (hzxuzhonghu) [inline ctl/monitoring/monitoring.go:66]: is this not a bool var?

Comment (LiZhenCheng9527) [reply ctl/monitoring/monitoring.go:66]: Using string is consistent with the rest of the command line.

File: ctl/monitoring/monitoring.go (hunk 8 above)
Comment (LiZhenCheng9527) [reply ctl/monitoring/monitoring.go:53]: Will return en error: `Error: unknown flag: --workloadMetric`

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**